components/script/dom/servoparser/mod.rs

```diff
@@ -736,6 +737,43 @@ impl FetchResponseListener for ParserContext {
             .and_then(|meta| meta.content_type)
             .map(Serde::into_inner)
             .map(Into::into);
+
+        // https://www.w3.org/TR/CSP/#initialize-document-csp
+        // TODO: Implement step 1 (local scheme special case)
+        let csp_list = metadata.as_ref().and_then(|m| {
+            let h = if let Some(h) = m.headers.as_ref() {
+                h
+            } else {
+                return None;
+            };
+            let mut csp = h.get_all("content-security-policy").iter();
+            // This silently ignores the CSP if it contains invalid Unicode.
+            // We should probably report an error somewhere.
```

Comment:
What doees the spec say about non-Unicode CSP headers?

Reply:
It says, https://www.w3.org/TR/CSP/#serialized-csp

> A serialized CSP is an *ASCII string* consisting of a semicolon-delimited series of serialized directives, adhering to the following ABNF grammar

I can't find any place in the spec where it says what to do if it isn't.